numpy/core/_internal.py

```diff
@@ -317,7 +275,13 @@ def data_as(self, obj):
 
         The returned pointer will keep a reference to the array.
         """
-        return self._ctypes.cast(self._data, obj)
+        # _ctypes.cast function causes a circular reference of self._data in
+        # self._data._objects. Attributes of self._data cannot be released
```

Comment:
This comment should reference the upstream python bug.

Reply:
Sure. The upstream CPython bug https://bugs.python.org/issue12836 was added in the comment.